plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/model/JDK.java

```diff
@@ -77,6 +83,30 @@ public String getCompatibleSince() {
         return compatibleSince;
     }
 
+    public String getLatestTestHarnessVersion() {
+        return latestTestHarnessVersion;
+    }
+
+    /**
+     * Get the latest compatible jenkins-test-harness version based on the Jenkins version.
+     * @param jenkinsVersion The Jenkins version.
+     * @return The latest compatible jenkins-test-harness version, or null.
+     */
+    public static String getLatestTestHarnessVersion(String jenkinsVersion) {
+        if (jenkinsVersion == null || jenkinsVersion.isEmpty()) {
+            throw new IllegalArgumentException("Jenkins version cannot be null or empty");
+        }
+
+        JDK oldestCompatibleJdk = min(Set.of(JDK.values()), jenkinsVersion);
+
+        // Return the latest test-harness version for the oldest compatible JDK
+        return oldestCompatibleJdk != null
+                ? oldestCompatibleJdk.getLatestTestHarnessVersion() == null
```

Comment:
I would keep returning null. Then the caller is responsible to do what's needed. For example use latest.release on openrewrite.